__tests__/app.test.js

```diff
@@ -113,10 +113,77 @@ describe('GET /api/topics', () => {
               votes: expect.any(Number),
               article_img_url: expect.any(String),
               comment_count: expect.any(String)
-            });
-          });
-        });
-    });
+            })
+          })
+        })
+    })
     
-  });
+  })
+
+  describe("GET /api/articles/:article_id/comments", () => {
+    test('200 OK: empty array when article_id is valid but has no comments', () => {
+      return request(app)
+        .get('/api/articles/2/comments')
+        .expect(200)
+        .then((response) => {
+          expect(response.body.comments).toEqual(expect.any(Array));
+        });
+        })
+
+    test("200 OK: all comments for an article", () => {
+      return request(app)
+        .get("/api/articles/1/comments")
+        .expect(200)
+        .then(({ body }) => {
+          const { comments } = body
+          expect(comments).toBeInstanceOf(Array)
+          expect(comments).toHaveLength(11) 
+          
+          comments.forEach((comment) => {
+            expect(comment).toHaveProperty("comment_id")
+            expect(comment).toHaveProperty("body")
+            expect(comment).toHaveProperty("votes")
+            expect(comment).toHaveProperty("author")
+            expect(comment).toHaveProperty("article_id")
+            expect(comment).toHaveProperty("created_at")
 
+            expect(comment).toMatchObject({
+              comment_id: expect.any(Number),
+              body:  expect.any(String),
+              votes: expect.any(Number),
+              author: expect.any(String),
+              article_id: expect.any(Number),
+              created_at: expect.any(String)
+            })
+          })
+      })
+  })
+  test('200 OK: comments served with the most recent comments first.', () => {
+    return request(app)
+      .get('/api/articles/1/comments')
+      .expect(200)
+      .then(({ body }) => {
+        const { comments } = body;
+        expect(comments[0].created_at).toBe("2020-11-03T21:00:00.000Z");
+        expect(comments[comments.length - 1].created_at).toBe("2020-01-01T03:08:00.000Z");
+      });
+  })
+
+    test("404 Not Found: valid api but no ID", () => {
+      return request(app)
+        .get("/api/articles/99999/comments")
+        .expect(404)
+        .then(({ body }) => {
+          expect(body.msg).toBe("Not Found")
+        })
+    })
+  
+    test("400 Bad Request: error message when passed invalid article ID", () => {
+      return request(app)
+        .get("/api/articles/noID/comments")
+        .expect(400)
+        .then(({ body }) => {
+          expect(body.msg).toBe("Bad Request")
+        })
+    })
+  })
```

Comment:
We're still missing the test here to check for a valid ID that exists but has no comments currently (200)